src/inmanta/ast/type.py

```diff
@@ -229,15 +229,21 @@ def __eq__(self, other: object) -> bool:
 @stable_api
 class Number(Primitive):
     """
-    This class represents an integer or float in the configuration model. On
-    these numbers the following operations are supported:
-
-    +, -, /, *
+    This class represents an integer or float in the configuration model.
     """
 
     def __init__(self) -> None:
         Primitive.__init__(self)
-        self.try_cast_functions: Sequence[Callable[[Optional[object]], numbers.Number]] = [int, float]
+        self.try_cast_functions: Sequence[Callable[[Optional[object]], numbers.Number]] = [float]
+
+    def cast(self, value: Optional[object]) -> object:
+        """
+        Attempts to cast a given value to an int or a float.
+        """
+        # Keep precision: cast to an int only if it already is an int
+        if isinstance(value, int):
+            return int(value)
```

Comment:
This case is a bit unnecessary

Reply:
The cast you mean? I recommended it because `isinstance(True, int)`, yet `int(True) is 1`. In other words: when you cast you really expect to get the numeric value, not a "number-compatible" one.

Reply:
Do we also allow boolean to be cast to int? Never realized.

Reply:
yes we do: https://github.com/inmanta/inmanta-core/blob/master/tests/compiler/test_typing.py#L113